### DISCUSSION TITLE
number of files per dir?

### DISCUSSION THREAD

**Discussion by JerryZ521** [minio/minio#17990] [Q&A] [answered]
os：ubuntu
run env：k8s
deployment mode：distributed，3 nodes

Now，one data dir has reached 100w+ files，it causes the cluster service unavailability。

problems：
1、node restart frequently，console log：disk not found...
2、access file：slow down... response code：503

**Comment by klauspost** [comment, marked as answer]
https://min.io/docs/minio/linux/administration/concepts.html#can-i-organize-objects-in-a-folder-structure-within-buckets

> Deployments with modest or budget-focused hardware should architect their workloads to target 10,000 objects per prefix as a baseline. Increase this target based on benchmarking and monitoring of real world workloads up to what the hardware can meaningfully handle.
